More respirate improvements

jeremyevans opened the pull request:
Improvements:

* For strands with expired leases, calculate delay based on lease time instead of schedule time, to prevent the delay numbers showing 120 second spikes when processing a strand with an expired lease.

* Make respirate sleep time dependent on dispatcher state.  This shouldn't do much in production currently, but it is necessary step to prevent starvation when decreasing the queue size ratio, which is something I'd like to try to see if it results in lower overall delay.
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Improve `respirate` by handling expired leases and dynamically adjusting sleep duration based on dispatcher state.
> 
>   - **Behavior**:
>     - Calculate delay based on lease time for expired leases in `Strand#respirate_metrics`.
>     - Adjust `respirate` sleep duration based on dispatcher state in `Dispatcher#sleep_duration`.
>   - **Config**:
>     - Change `dispatcher_queue_size_ratio` type from `int` to `float` in `config.rb`.
>   - **Metrics**:
>     - Add `lease_expired_percentage` to metrics in `Dispatcher#metrics_hash`.
>   - **Tests**:
>     - Add tests for expired lease handling in `dispatcher_spec.rb`.
>     - Add tests for dynamic sleep duration in `dispatcher_spec.rb`.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=ubicloud%2Fubicloud&utm_source=github&utm_medium=referral)<sup> for 155770cebd12c2881ec9c1739fa8468ad32692c6. You can [customize](https://app.ellipsis.dev/ubicloud/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

fdr reviewed (approved):
alternative abstract: adaptive scan polling, I suppose?

jeremyevans commented:
Rebased and removed the "Do not push to the metrics queue if shutting down", as that was handled a different way in 1fcb10a6a291373d7be7b8344cdc24b365e9a997.
